src/include/cluster/dialogs.rb

```diff
@@ -708,6 +708,205 @@ def SaveSecurity
+      Cluster.qdevice_port = UI.QueryWidget(Id(:qdevice_port), :Value).to_s
+      Cluster.qdevice_tls = UI.QueryWidget(Id(:qdevice_tls), :Value)
+      Cluster.qdevice_algorithm = UI.QueryWidget(Id(:qdevice_algorithm), :Value)
+      Cluster.qdevice_tie_breaker = UI.QueryWidget(Id(:qdevice_tie_breaker), :Value)
+      nil
+    end
+
+    def CorosyncQdeviceLayout
+      qdevice_section = VBox(
+        HBox(
+          ComboBox(
+            Id(:qdevice_model),
+            Opt(:hstretch),
+            _("Qdevice model:"),
+            ["net"]
+          ),
+          Left(InputField(Id(:qdevice_votes),Opt(:hstretch), _("Qdevice votes:"),""))
+        )
+      )
+
+      qdevice_net_section = VBox(
+        HBox(
+          Left(InputField(Id(:qdevice_host),Opt(:hstretch), _("Qnetd server host:"),"")),
+          HSpacing(1),
+          Left(InputField(Id(:qdevice_port),Opt(:hstretch), _("Qnetd server TCP port:"),"5403"))
+        ),
+        HBox(
+          Left(ComboBox(
+            Id(:qdevice_tls), Opt(:hstretch), _("TLS:"),
+            ["off", "on", "required"]
+          )),
+          Left(ComboBox(
+            Id(:qdevice_algorithm), Opt(:hstretch, :notify), _("Algorithm:"),
+            [
+            Item(Id("ffsplit"),"ffsplit"),
+            Item(Id("lms"),"lms")
+            ]
+          )),
+          HSpacing(1),
+          Left(InputField(Id(:qdevice_tie_breaker),Opt(:hstretch), _("Tie breaker:"),"lowest"))
+        )
+      )
+
+      contents = VBox(
+        VSpacing(1),
+        CheckBoxFrame(
+          Id(:corosync_qdevice),
+          Opt(:hstretch, :notify),
+          _("En&able Corosync Qdevice"),
+          false,
+          VBox(
+            VSpacing(1),
+            qdevice_section,
+            VSpacing(2),
+            qdevice_net_section,
+          )
+        ),
+        VStretch()
+      )
+
+      my_SetContents("corosyncqdevice", contents)
+
+      nil
+    end
+
+    def UpdateQdeviceVotes
+      is_ffsplit = UI.QueryWidget(Id(:qdevice_algorithm), :Value) == "ffsplit"
+
+      if is_ffsplit
+        UI.ChangeWidget(Id(:qdevice_votes), :Value, "1")
+      end
+
+      UI.ChangeWidget(Id(:qdevice_votes), :Enabled, !is_ffsplit)
+
+      nil
+    end
+
+    def CorosyncQdeviceDialog
+      ret = nil
+
+      CorosyncQdeviceLayout()
+
+      UI.ChangeWidget(Id(:corosync_qdevice), :Value, Cluster.corosync_qdevice)
+
+      UI.ChangeWidget(Id(:qdevice_model), :Value, Cluster.qdevice_model)
+      UI.ChangeWidget(Id(:qdevice_votes), :Value, Cluster.qdevice_votes)
+      UI.ChangeWidget(Id(:qdevice_votes), :ValidChars, "0123456789");
+
+      UI.ChangeWidget(Id(:qdevice_host), :Value, Cluster.qdevice_host)
+      UI.ChangeWidget(Id(:qdevice_port), :Value, Cluster.qdevice_port)
+      UI.ChangeWidget(Id(:qdevice_tls), :Value, Cluster.qdevice_tls)
+      UI.ChangeWidget(Id(:qdevice_algorithm), :Value, Cluster.qdevice_algorithm)
+      UI.ChangeWidget(Id(:qdevice_tie_breaker), :Value, Cluster.qdevice_tie_breaker)
```

Comment:
what about having init method for whole this initialization?

Reply:
Move the code to function CorosyncQdeviceLayout(). It is mainly about initialization, including draw the UI and initialize the parameters.